app/controllers/katello/api/v2/products_controller.rb

```diff
@@ -41,8 +41,7 @@ def index_relation
       query = query.where(:name => params[:name]) if params[:name]
       query = query.enabled if params[:enabled]
       query = query.where(:id => @activation_key.products) if @activation_key
-      query = query.where(:id => @system.products) if @system
-      query = query.where(:id => Pool.find_by_id!(params[:subscription_id]).products) if params[:subscription_id]
+      query = query.where(:id => Pool.find_by_cp_id(params[:subscription_id]).products) if params[:subscription_id]
```

Comment:
the front end is passing the candlepin id, so I decided to keep that, but yes I agree, are you thinking we should be able to pass the database id as well?